src/ApplicationLogs/LogReader.cs

```diff
@@ -8,38 +8,55 @@
 // Redistribution and use in source and binary forms with or without
 // modifications are permitted.
 
+using ApplicationLogs.Store;
+using ApplicationLogs.Store.Models;
+using Neo.ConsoleService;
 using Neo.IO;
 using Neo.Json;
 using Neo.Ledger;
 using Neo.Network.P2P.Payloads;
 using Neo.Persistence;
 using Neo.SmartContract;
+using Neo.SmartContract.Native;
 using Neo.VM;
-using System;
-using System.Collections.Generic;
-using System.Linq;
+using System.Numerics;
 using static System.IO.Path;
 
 namespace Neo.Plugins
 {
     public class LogReader : Plugin
     {
-        private IStore _db;
-        private ISnapshot _snapshot;
+        #region Globals
+
+        private NeoStore _neostore;
+        private NeoSystem _neosystem;
+        private List<LogEventArgs> _logEvents;
+
+        #endregion
 
         public override string Name => "ApplicationLogs";
-        public override string Description => "Synchronizes the smart contract log with the NativeContract log (Notify)";
+        public override string Description => "Synchronizes smart contract VM executions and notificatons (NotifyLog) on blockchain.";
+
+        #region Ctor
 
         public LogReader()
         {
+            _logEvents = new();
             Blockchain.Committing += OnCommitting;
             Blockchain.Committed += OnCommitted;
         }
 
+        #endregion
+
+        #region Override Methods
+
         public override void Dispose()
         {
             Blockchain.Committing -= OnCommitting;
             Blockchain.Committed -= OnCommitted;
+            if (Settings.Default.Debug)
+                ApplicationEngine.Log -= OnApplicationEngineLog;
+            GC.SuppressFinalize(this);
```

Comment:
In this safe, everything was disposed? No memory leak?

Reply:
No memory leaks. But i do see something that needs changing.

Reply:
nevermind, its all good.